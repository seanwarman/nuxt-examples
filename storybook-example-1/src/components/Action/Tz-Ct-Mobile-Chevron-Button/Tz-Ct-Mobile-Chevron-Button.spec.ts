/**
 * @vitest-environment jsdom
 */

import { mount, VueWrapper } from '@vue/test-utils';
import { expect, describe, it } from 'vitest';
import TzCtMobileChevronButton from './Tz-Ct-Mobile-Chevron-Button.vue';

let wrapper: VueWrapper;

function makeGetClasses(wrapper: VueWrapper) {
  return (selector: string) => {
    return wrapper.get(selector).classes();
  };
}

const iconSelector = '[data-test="tz-ct-chevron-button"]';

describe('Chevron button setup', () => {
  it('renders correctly', () => {
    wrapper = mount(TzCtMobileChevronButton, {
      props: {
        direction: 'left'
      }
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders a chevron pointing right', () => {
    wrapper = mount(TzCtMobileChevronButton, {
      props: {
        direction: 'right'
      }
    });

    const getClasses = makeGetClasses(wrapper);

    expect(getClasses(iconSelector)).not.toContain('rotate-180');
  });

  it('renders a chevron pointing left', () => {
    wrapper = mount(TzCtMobileChevronButton, {
      props: {
        direction: 'left'
      }
    });

    const getClasses = makeGetClasses(wrapper);

    expect(getClasses(iconSelector)).toContain('rotate-180');
  });

  it("emits a 'click' event when it is clicked", () => {
    wrapper.find('button').trigger('click');
    expect(wrapper.emitted()).toHaveProperty('click');
  });
});
