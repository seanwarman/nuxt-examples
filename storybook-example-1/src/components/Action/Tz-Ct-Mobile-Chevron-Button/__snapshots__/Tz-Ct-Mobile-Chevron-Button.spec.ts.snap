// Vitest Snapshot v1

exports[`Chevron button setup > renders correctly 1`] = `
<button
  class="grid h-16 w-16 place-items-center bg-dark text-white ring-dark hover:bg-black focus:ring"
>
  <svg
    class="w-5 rotate-180"
    data-test="tz-ct-chevron-button"
    fill="none"
    viewBox="0 0 32 32"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M19.775 16 4.555 7.629l2.89-5.258L24.26 11.62c3.454 1.9 3.454 6.863 0 8.762L7.447 29.63l-2.893-5.26L19.774 16Z"
      fill="currentColor"
      fill-rule="evenodd"
    />
  </svg>
</button>
`;
